src/main/java/com/excelsiorjet/api/tasks/JetProject.java

```diff
@@ -643,13 +641,116 @@ public JetHome validate(boolean validateForBuild) throws JetTaskFailureException
             execProfilesName = projectName;
         }
         if (validateForBuild) {
-            validateForBuild(jetHomeObj);
+            validateForBuild(excelsiorJet);
         }
 
-        return jetHomeObj;
+        processDependencies();
     }
 
-    private void validateForBuild(JetHome jetHomeObj) throws JetTaskFailureException {
+    public void processDependencies() throws JetTaskFailureException {
+        for (DependencySettings dependency : dependencies) {
+            if (!dependency.isValid()) {
+                throw new JetTaskFailureException(s("JetApi.InvalidDependencySetting", dependency.idStr()));
+            }
+        }
+        List<DependencySettings> dependenciesSettings = dependencies.stream()
+                .filter(DependencySettings::isSettings)
```

Comment:
I do not like two predicates with opposite meanings isSettings and isExternal. There is no evidence from the code that you do not miss some dependencies when you filter by the predicates. I would remain isExternal only